components/docker-auth/cmd/authn/authentication.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/golang/glog"
+
+	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg"
+)
+
+var (
+	idpKey        []byte
+	dockerAuthKey []byte
+	err           error
+)
+
+const (
+	issuerClaim = "iss"
+	subjectClaim = "sub"
+	expirationClaim = "exp"
+	idpCertEnvVar = "IDP_CERT"
+	authTokenIssuer = "REGISTRY_AUTH_TOKEN_ISSUER"
+	dockerAuthCertEnvVar = "REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE"
+)
+
+func errLog(err error) {
+	if err != nil {
+		log.Fatal("Error:", err.Error())
+	}
+}
+
+func init() {
+	flag.Usage = usage
+	flag.Parse()
+	idpKey, err = ioutil.ReadFile(os.Getenv(idpCertEnvVar))
+	if err != nil {
+		glog.Info("unable to find the idp key")
+		glog.Flush()
+		os.Exit(pkg.ErrorExitCode)
+	}
+	dockerAuthKey, err = ioutil.ReadFile(os.Getenv(dockerAuthCertEnvVar))
+	if err != nil {
+		glog.Info("unable to find the docker auth key")
+		glog.Flush()
+		os.Exit(pkg.ErrorExitCode)
+	}
+}
+
+func usage() {
+	flag.PrintDefaults()
+	os.Exit(pkg.MisuseExitCode)
+}
+
+func getJWTClaim(token interface{}, claimKey string) string {
+	jwtToken, _ := jwt.Parse(token.(string), nil)
+	claims, _ := jwtToken.Claims.(jwt.MapClaims)
+	return claims[claimKey].(string)
+}
+
+func SignatureValidate(inToken interface{}, cert []byte) (interface{}, error) {
```

Comment:
This isn't used by anything else outside this file. So can we make this private ?